react/src/Sidebar/SidebarList.tsx

```diff
@@ -54,6 +55,37 @@ export interface SidebarListProps extends BaseSidebarItemProps {
   onClick?: () => void
 }
 
+type SectionWrapperProps = HTMLChakraProps<'button'> &
+  HTMLChakraProps<'div'> & { onlyCaretToggle: boolean }
+
+const SectionWrapper: FC<PropsWithChildren & SectionWrapperProps> = ({
+  children,
+  onlyCaretToggle,
+  ...props
+}) => {
+  if (onlyCaretToggle) return <chakra.div {...props}>{children}</chakra.div>
+
+  return (
+    <chakra.button {...props} type="button">
```

Comment:
should we set the `type` declaration to be before `props` destructure? in case props itself has a type override

```suggestion
    <chakra.button type="button" {...props}>
```